Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pickable.py

```diff
@@ -50,48 +90,102 @@ def parser(cls, parser, loggers=None):
             dest='json',
             default=False,
         )
+        parser.add_argument(
+            '--into', '-i',
+            help='Branch to pick changes into',
+            type=str,
+            dest='into',
+            default=None,
+        )
+        parser.add_argument(
+            '--exclude', '-e',
+            help='Exclude certain patterns. By default, assumes the string argument is a regex. The program '
+                'will use certain prepared regexes for some string arguments (some of which are enabled by default), '
+                'those arguments are: {}'.format(string_utils.join([
+                    "'{}' ({})".format(pattern, 'enabled' if pattern in cls.DEFAULT_PATTERNS else 'disabled')
+                    for pattern in cls.PATTERNS.keys()
+                ])),
+            action='append',
+            dest='excluded',
+            default=None,
+        )
+        parser.add_argument(
+            '--scope', '-s',
+            help='Scope pickable query to a specific path in the project',
+            action='append',
+            dest='scopes',
+            default=None,
+        )
+        parser.add_argument(
+            '--filter', '-f',
+            help='Print only commits whose message matches a provided filter',
+            action='append',
+            dest='filters',
+            default=[],
+        )
+        parser.add_argument(
+            '--by', '-b',
+            help='Print only commits made by a specific individual (or individuals)',
+            action='append',
+            dest='by',
+            default=[],
+        )
 
     @classmethod
-    def pickable(cls, commits, repository, commits_story=None):
+    def pickable(cls, commits, repository, commits_story=None, excluded=None):
         filtered_in = set()
         all_commits = dict()
 
         commits_story = commits_story or CommitsStory()
 
+        if excluded is None:
+            excluded = cls.DEFAULT_PATTERNS
+        excluded = list(itertools.chain(
+            *[cls.PATTERNS[arg] if arg in cls.PATTERNS else [re.compile(arg)] for arg in excluded]
+        ))
+
         for commit in commits:
             commits_story.add(commit)
             title = commit.message.splitlines()[0]
             all_commits[str(commit)] = commit
-            if search('Cherry-pick', title) or search('Versioning', title):
+            if any([ex(title) if callable(ex) else ex.search(title) for ex in excluded]):
```

Comment:
If we construct excluded the way mentioned above, this can just be `if any(ex(title) for ex in excluded)`